### PR TITLE
feat: update get seedlots by user id to get sl by client

### LINKED ISSUE #1092: Update get seedlots by user_id to get sl by client_id

### PR DESCRIPTION
**Pull request by RMCampos**
# Description
Closes #1092 

#### New
- None;

#### Changed
- Seedlot `Endpoint`, `Service` and `Repository` classes enabling users to fetch seedlots by client id instead of user id.

#### Removed
- None;

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [x] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-28-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1178-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1178-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-28-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1178-backend.apps.silver.devops.gov.bc.ca/actuator/health)
- [Oracle-API](https://nr-spar-1178-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by mgaseta — APPROVED** [review]
LGTM! 
Good job :100: :1st_place_medal:

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java [hunk 1]
```diff
@@ -193,15 +195,20 @@ private String nextSeedlotNumber(Character seedlotClassCode) {
...
+   * @param clientId the id of the client to fetch the seedlots for
    * @param pageNumber the page number for the paginated search
    * @param pageSize the size of the page
    * @return a list of the user's seedlots
    */
-  public Optional<Page<Seedlot>> getUserSeedlots(String userId, int pageNumber, int pageSize) {
-    SparLog.info("Retrieving paginated list of seedlots for the user {}", userId);
+  public Optional<Page<Seedlot>> getUserSeedlots(String clientId, int pageNumber, int pageSize) {
+    Optional<UserInfo> userInfo = loggedUserService.getLoggedUserInfo();
+    if (userInfo.isPresent() && !userInfo.get().clientIds().contains(clientId)) {
```
Comment (RMCampos) [inline backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java:207]: @craigyu here's the validation you asked for :)

**Review by craigyu — APPROVED** [review]
Works great! ezpz for you

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java (hunk 1 above)
Comment (craigyu) [reply backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java:207]: this is exactly it!

**Comment by craigyu** [comment]
This will make the front-end stop working with the my-seedlots page however, I can make a quick PR for it Wednesday morning, wait for me? perhaps we can use a staging branch again

**Pull request merged**